Update remote-development.md (#17726)
docker/docs: develop/remote-development.md

This takes you to a step-by-step guide on setting up Telepresence,
connecting to a development cluster, and creating intercepts.

## mirrord

mirrord is an open-source tool that lets developers run local processes in the context of their cloud environment. When mirrord is executed, it injects itself into your local process, and starts a temporary pod in your Kubernetes cluster called the mirrord Agent. It then overrides low level functions in your local process, and relays them to a pod in your Kubernetes cluster through the mirrord Agent. mirrord does this for incoming and outgoing traffic, file access, and environment variables, so your process "thinks" it's running in the pod you selected, even though it's actually running locally.

![High-level architecture of mirrord](https://mirrord.dev/docs/overview/architecture/architecture.svg)

> Everything mirrord does can be configured to run either locally or remotely and so suit any type of use case. Incoming traffic supports the following modes:

Reviewer: ```suggestion
You can configure mirrord to run either locally or remotely and suit any type of use case. Incoming traffic supports the following modes:
```

Author: What I was trying to say is that mirrord's specific functions can be tweaked to work either locally or remotely (e.g. read files from your local machine, get environment variables from the remote pod, etc.). I changed the phrasing, let me know if it works now.

Author: Hi @ChrisChinchilla , wanted to ask if this is still under review?
